### PR TITLE
Feat/civ 2669 notification of hearing

### PR DESCRIPTION
**Pull request by hmcts-version1-ignacio**
### When an SDO has been made, and a hearing date has been setup a notification needs to be sent to the claimant to login to review the case, hearing date and where applicable the hearing fee ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: bin/add-roles.sh [hunk 1]
```diff
@@ -18,6 +18,7 @@
 ./bin/utils/ccd-add-role.sh "caseworker-ras-validation"
 ./bin/utils/ccd-add-role.sh "admin-access"
 ./bin/utils/ccd-add-role.sh "full-access"
+./bin/utils/ccd-add-role.sh "hearing-schedule-access"
 ./bin/utils/ccd-add-role.sh "civil-administrator-standard"
 ./bin/utils/ccd-add-role.sh "civil-administrator-basic"
 ./bin/utils/ccd-add-role.sh "hearing-schedule-access"
```
Comment (m-meulendijks-v1) [inline bin/add-roles.sh:24]: Does this role need to be in here twice? I think this is an accidental conflict resolution error :)

**Review by m-meulendijks-v1 — APPROVED** [review]
Approved, apart from little comment about duplication that is easily resolved.

File: bin/add-roles.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@
 ./bin/utils/ccd-add-role.sh "caseworker-ras-validation"
 ./bin/utils/ccd-add-role.sh "admin-access"
 ./bin/utils/ccd-add-role.sh "full-access"
+./bin/utils/ccd-add-role.sh "hearing-schedule-access"
```
Comment (m-meulendijks-v1) [inline bin/add-roles.sh:21]: My comment disappeared, how odd. I think this line is in there already at line 24.

File: bin/add-roles.sh (hunk 1 above)
Comment (hmcts-version1-ignacio) [reply bin/add-roles.sh]: Thank you! solved

**Review by m-meulendijks-v1 — APPROVED** [review]

File: charts/civil-service/values.aat.template.yaml [hunk 3]
```diff
@@ -49,6 +49,7 @@ java:
         - launch-darkly-sdk-key
         - robotics-notification-multipartyrecipient
         - ordnance-survey-api-key
+        - notification-of-hearing
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply charts/civil-service/values.aat.template.yaml]: I think not, will test it. Thank you!

File: charts/civil-service/values.preview.template.yaml [hunk 4]
```diff
@@ -48,6 +48,7 @@ java:
         - launch-darkly-sdk-key
         - robotics-notification-multipartyrecipient
         - ordnance-survey-api-key
+        - notification-of-hearing
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply charts/civil-service/values.preview.template.yaml]: I think not, will test it. Thank you!

**Review by filippo-possenti-version1 — DISMISSED** [review]
All fine by me.

There's a couple of minor improvements if you want, but no deal-breaker.

I would suggest a thorough testing against our test scenarios before we release to QA.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationDefendantOfHearingHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    class AboutToSubmitCallback {
+
+        @BeforeEach
+        void setup() {
+            when(notificationsProperties.getHearingListedNoFeeDefendantLrTemplate())
+                .thenReturn("test-template-no-fee-defendant-id");
+        }
+
+        @Nested
+        class OneVsOneScenario {
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationDefendantOfHearingHandlerTest.java:73]: Yes, thank you!

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationClaimantOfHearingHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+        @BeforeEach
+        void setup() {
+            when(notificationsProperties.getHearingListedFeeClaimantLrTemplate())
+                .thenReturn("test-template-fee-claimant-id");
+            when(notificationsProperties.getHearingListedNoFeeClaimantLrTemplate())
+                .thenReturn("test-template-no-fee-claimant-id");
+        }
+
+        @Nested
+        class OneVsOneScenario {
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationClaimantOfHearingHandlerTest.java:59]: Yes, thank you!

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationDefendantOfHearingHandler.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        } else {
+            String recipient2 = caseData.getRespondentSolicitor2EmailAddress();
+            sendEmail(caseData, recipient, true);
+            sendEmail(caseData, recipient2, false);
+        }
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .build();
+    }
+
+    private void sendEmail(CaseData caseData, String recipient, Boolean isFirst) {
```
[the remark this answers is not in the document]
Comment (hmcts-version1-ignacio) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationDefendantOfHearingHandler.java:63]: Good point, also it takes less memory. Thank you!

**Review by filippo-possenti-version1 — APPROVED** [review]
Nothing obvious I could spot here.

Provided that we tested locally against the Acceptance Criteria, this can be moved to QA.

**Pull request closed without merging**